server/build.gradle

```diff
@@ -124,6 +124,10 @@ dependencies {
   api "com.google.protobuf:protobuf-java:${versions.protobuf}"
   api "jakarta.annotation:jakarta.annotation-api:${versions.jakarta_annotation}"
 
+  // roaring bitmaps
+  api 'org.roaringbitmap:RoaringBitmap:0.9.49'
+  runtimeOnly 'org.roaringbitmap:shims:0.9.49' // might fix complaining about ArraysShims?
```

Comment:
I'm not sure that we have enough justification to add another dependency to core for this.

Given that we're going to have false positives, could we use a Bloom filter? Alternatively, we could use `RoaringDocIdSet`, which is already provided by Lucene? Or we could use `BytesRefHash`.

The only advantage that RBM has over those is that it supports a `remove` operation, though it's an expensive operation (that involves rebuilding the array for a given container).

From a design standpoint, I question whether we would even need to support a `remove` operation in a tiered cache. A disk-based cache could support eviction of entire "chunks", in which case you would just discard the whole chunk.